### PR TITLE
Fixing the key light pass with ambient map

### PR DESCRIPTION
**Pull request by samcake**
the key light pass was using the wrong program.
this is now fixed
THere is still a problem with the sampler filter used that seems to be nearest (visible when looking at a rough surface), i will fix it in a future PR

### CODE REVIEW

**Comment by samcake** [comment]
Test PLan:
The ambient map should be back to normal on the metallic and reflective material
Performance shoudn't be affected

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 1]
```diff
@@ -487,10 +487,8 @@ void RenderDeferredSetup::run(const render::SceneContextPointer& sceneContext, c
                 }
             } else {
                 if (keyLight->getAmbientMap()) {
-                    program = deferredLightingEffect->_directionalAmbientSphereLight;
-                    locations = deferredLightingEffect->_directionalAmbientSphereLightLocations;
-                    //program = deferredLightingEffect->_directionalSkyboxLight;
-                    //locations = deferredLightingEffect->_directionalSkyboxLightLocations;
+                    program = deferredLightingEffect->_directionalSkyboxLight;
```
Comment (ZappoMan) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: this logic is confusing... 

`if (keyLight->getAmbientMap())` 
*then* 
     ` program = deferredLightingEffect->_directionalSkyboxLight;`
*else*
     ` program = deferredLightingEffect->_directionalAmbientSphereLight;`

can you add a comment explaining why if there is not an ambient map we use the ambient light?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7af3aa53acd75f5668fc146db1a4644df51bf1b6 available at: https://gist.github.com/cba259366ed8883d653bf99235bdfea4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8782-7af3aa5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8782-7af3aa5.dmg

File: libraries/render-utils/src/DeferredLightingEffect.cpp (hunk 1 above)
Comment (samcake) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: just did

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7ea790aa4fd06237fa0ce09785328dfb2b7f7aa4 available at: https://gist.github.com/cba259366ed8883d653bf99235bdfea4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8782-7ea790a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8782-7ea790a.dmg

**Comment by AlanZimmerman** [comment]
Tested this on Windows. Ambient map reflections appear to work as the were working before. No noticeable hit on performance in spaces I've been in in the RC today. Also tested in the HMD--all good.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3ebab3b017a0d86530d13001e22e65064ea5c390 available at: https://gist.github.com/cba259366ed8883d653bf99235bdfea4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8782-3ebab3b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8782-3ebab3b.dmg

**Pull request merged**